src/Datastore/DatastoreClient.php

```diff
@@ -0,0 +1,796 @@
+     *
+     * Example:
+     * ```
+     * $key = $datastore->key('Person');
+     * $keyWithAllocatedId = $datastore->allocateId($key);
+     * ```
+     *
+     * @see https://cloud.google.com/datastore/reference/rest/v1beta3/projects/allocateIds allocateIds
+     *
+     * @param Key $key The incomplete key
+     * @param array $options Configuration options
+     * @return Key
+     */
+    public function allocateId(Key $key, array $options = [])
+    {
+        $res = $this->allocateIds([$key], $options);
+        return $res[0];
+    }
+
+    /**
+     * Allocate available IDs to a set of keys
+     *
+     * Keys MUST be in an incomplete state (i.e. including a kind but not an ID
+     * or name in their final pathElement).
+     *
+     * This method will execute a service request.
+     *
+     * Example:
+     * ```
+     * $keys = [
+     *     $datastore->key('Person'),
+     *     $datastore->key('Person')
+     * ];
+     *
+     * $keysWithAllocatedIds = $datastore->allocateIds($keys);
+     * ```
+     *
+     * @see https://cloud.google.com/datastore/reference/rest/v1beta3/projects/allocateIds allocateIds
+     *
+     * @param Key[] $keys The incomplete keys
+     * @param array $options Configuration options
+     * @return Key[]
+     */
+    public function allocateIds(array $keys, array $options = [])
+    {
+        return $this->operation->allocateIds($keys, $options);
+    }
+
+    /**
+     * Create a Transaction
+     *
+     * Example:
+     * ```
+     * $transaction = $datastore->transaction();
+     * ```
+     *
+     * @see https://cloud.google.com/datastore/docs/concepts/transactions Datastore Transactions
+     *
+     * @param array $options Configuration options
+     * @return Transaction
+     */
+    public function transaction(array $options = [])
+    {
+        $res = $this->connection->beginTransaction($options + [
+            'projectId' => $this->projectId
+        ]);
+
+        return new Transaction(
+            clone $this->operation,
+            $this->projectId,
+            $res['transaction']
+        );
+    }
+
+    /**
+     * Insert an entity
+     *
+     * An entity with incomplete keys will be allocated an ID prior to insertion.
+     *
+     * Example:
+     * ```
+     * $key = $datastore->key('Person', 'Bob');
+     * $entity = $datastore->entity($key, ['firstName' => 'Bob']);
+     *
+     * $datastore->insert($entity);
+     * ```
+     *
+     * @param Entity $entity The entity to be inserted
+     * @param array $options Configuration options
+     * @return string The entity version
+     * @throws DomainException If a conflict occurs, fail.
+     */
+    public function insert(Entity $entity, array $options = [])
+    {
+        $res = $this->insertBatch([$entity], $options);
+        return $this->parseSingleMutationResult($res);
+    }
+
+    /**
+     * Insert multiple entities
+     *
+     * Any entity with incomplete keys will be allocated an ID prior to insertion.
```

Comment:
Insertion in this case would be non-transactional, right? If so, it may be helpful to document.